cypress/integration/admin_portal/tests/test_code_management.spec.js

```diff
@@ -1,55 +1,163 @@
+    codeManagementDashboard.getLabels('Codes').should('have.text', 'Number of Codes')
+    codeManagementDashboard.getRequestCodesButton().should('have.attr', 'type', 'submit').and('have.text', 'Request Codes')
+    codeManagementDashboard.getCancelButton().should('have.attr', 'href', '/successfactors/admin/coupons').and('have.text', 'Cancel')
+    codeManagementDashboard.getFormField('emailAddress').clear().prev().click()
+    codeManagementDashboard.getInvalidFeedback().should('have.text', 'This field is required.')
+    codeManagementDashboard.getFormField('emailAddress').type('zrana@')
+    codeManagementDashboard.getInvalidFeedback().should('have.text', 'Must be a valid email address.')
+    codeManagementDashboard.getRequestCodesButton().should('have.attr', 'disabled')
+   })
 
   it('Verifies the correct functioning of REQUEST CODES form', function () {
-   const request_codes_url = 'https://portal.stage.edx.org/successfactors/admin/coupons/request'
-   const code_management_main = 'https://portal.stage.edx.org/successfactors/admin/coupons'
-   codeManagementDashboard.requestMoreCodes()
-   cy.url().should('eq', request_codes_url)
-   codeManagementDashboard.getCancelButton().click()
-   cy.url().should('eq', code_management_main)
-   codeManagementDashboard.requestMoreCodes()
-   cy.url().should('eq', request_codes_url)
-  //  Request for more codes
-   codeManagementDashboard.getFormField('numberOfCodes').type('5')
-   codeManagementDashboard.getRequestCodesButton().click()
-   cy.url().should('eq', code_management_main)
-  //  Cypress returns by concatenating both the strings, so we are asserting both success messages as substrings
-   codeManagementDashboard.getSuccessMessage().should(($dialogue_message)=>{
-     const text = $dialogue_message.text()
-     expect(text).includes('Request for more codes received')
-     expect(text).includes('The edX Customer Success team will contact you soon.')
+    const request_codes_url = 'https://portal.stage.edx.org/successfactors/admin/coupons/request'
+    const code_management_main = 'https://portal.stage.edx.org/successfactors/admin/coupons'
+    codeManagementDashboard.requestMoreCodes()
+    cy.url().should('eq', request_codes_url)
+    codeManagementDashboard.getCancelButton().click()
+    cy.url().should('eq', code_management_main)
+    codeManagementDashboard.requestMoreCodes()
+    cy.url().should('eq', request_codes_url)
+   //  Request for more codes
+    codeManagementDashboard.getFormField('numberOfCodes').type('5')
+    codeManagementDashboard.getRequestCodesButton().click()
+    cy.url().should('eq', code_management_main)
+   //  Cypress returns by concatenating both the strings, so we are asserting both success messages as substrings
+    codeManagementDashboard.getSuccessMessage().should(($dialogue_message)=>{
+      const text = $dialogue_message.text()
+      expect(text).includes('Request for more codes received')
```

Comment:
save all strings as const preferably in dictionary form or on top so if any message changes we don't have to search it in code